server/sonar-server/src/test/java/org/sonar/server/user/ws/HomepageFinderTest.java

```diff
@@ -0,0 +1,119 @@
+
+public class HomepageFinderTest {
+
+  @Rule
+  public UserSessionRule userSessionRule = UserSessionRule.standalone();
+  @Rule
+  public DbTester dbTester = DbTester.create(System2.INSTANCE);
+  @Rule
+  public ExpectedException expectedException = ExpectedException.none();
+
+  private DbClient dbClient = dbTester.getDbClient();
+  private DbSession dbSession = dbTester.getSession();
+
+  @Test
+  public void find_project_homepage_for_users_that_have_one() throws Exception {
+    OrganizationDto organization = dbTester.organizations().insert();
+    ComponentDto project = ComponentTesting.newPublicProjectDto(organization);
+    dbTester.components().insertComponent(project);
+
+    HomepageFinder underTest = new HomepageFinder(dbClient);
+
+    Homepage homepage = underTest.findFor(dbSession, new UserDto().setHomepageType("PROJECT").setHomepageKey(project.uuid()));
+
+    assertThat(homepage).isNotNull();
+    assertThat(homepage.getType()).isEqualTo(PROJECT);
+    assertThat(homepage.getValue()).isEqualTo(project.getKey());
+  }
+
+  @Test
+  public void find_organization_homepage_for_users_that_have_one() throws Exception {
+    OrganizationDto organization = dbTester.organizations().insert();
+
+    HomepageFinder underTest = new HomepageFinder(dbClient);
+
+    Homepage homepage = underTest.findFor(dbSession, new UserDto().setHomepageType("ORGANIZATION").setHomepageKey(organization.getUuid()));
+
+    assertThat(homepage).isNotNull();
+    assertThat(homepage.getType()).isEqualTo(ORGANIZATION);
+    assertThat(homepage.getValue()).isEqualTo(organization.getKey());
+  }
+
+
+  @Test
+  @Ignore // Default WIP implementation to be done in SONAR-10185
+  public void default_if_homepage_project_key_does_not_match_a_project() throws Exception {
+    String frozenUUID = UUID.randomUUID().toString();
+    UserDto userDto = new UserDto().setHomepageType("PROJECT").setHomepageKey(frozenUUID);
+
+    HomepageFinder underTest = new HomepageFinder(dbClient);
+
+    Homepage homepage = underTest.findFor(dbSession, userDto);
+    assertThat(homepage).isNotNull();
+  }
+
+  @Test
+  @Ignore // Default WIP implementation to be done in SONAR-10185
+  public void default_if_homepage_organization_key_does_not_match_an_organization() throws Exception {
+    String frozenUUID = UUID.randomUUID().toString();
+    UserDto userDto = new UserDto().setHomepageType("ORGANIZATION").setHomepageKey(frozenUUID);
+
+    HomepageFinder underTest = new HomepageFinder(dbClient);
+
+    expectedException.expect(IllegalStateException.class);
+    expectedException.expectMessage("No Organization found for homepage key " + frozenUUID);
+
+    Homepage homepage = underTest.findFor(dbSession, userDto);
+    assertThat(homepage).isNotNull();
+  }
+
+
+  @Test
+  @Ignore // Default WIP implementation to be done in SONAR-10185
+  public void find_default_homepage_when_users_does_not_have_one() throws Exception {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Fix or remove this skipped unit test [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1607)